collections/_article/upcoming-serious-web-performance-boost.md

```diff
@@ -0,0 +1,68 @@
+---
+title: "Upcoming (serious) Web performance boost"
+excerpt: "Compiling the Godot Engine with WASM SIMD support truly is a game changer."
+categories: ["progress-report"]
+author: Adam Scott
+image: /storage/blog/covers/upcoming-serious-web-performance-boost.webp
+date: 2025-05-21 18:00:00
+---
+
+Sometimes, just adding a compiler flag can yield significant performance boosts. [And that just happened.](https://github.com/godotengine/godot/pull/106319)
+
+[For about two years now](https://caniuse.com/wasm-simd), all major browsers have supported WASM ([WebAssembly](https://en.wikipedia.org/wiki/WebAssembly)) SIMD. [SIMD stands for "Single instruction, multiple data".](https://en.wikipedia.org/wiki/Single_instruction,_multiple_data) It is a technology that permits CPUs to do some parallel computation, often speeding up the whole program. And that's exactly why we tried it out recently.
+
+We got positive results.
+
+## The need for performance on the Web
+
+The Web platform is often considered the exports' poor stepchild, because of it's less-than-ideal environment and it's percieved poor performance. And the perception is somewhat right: the Web environment has a lot of security-related quirks to take into account—the user needs to interact with a game frame before the browser allows it to play any sound[^1]. Also, due to bandwidth and compatibility-reasons, you rarely see high-fidelity games being played on a browser. Performance is better achieved when running software natively on the operating system.
+
+But don't underestimate the potential of the Web platform. As I explained in broad terms at the talk I gave at the last GodotCon Boston 2025, the Web has caught up a lot since the days of Flash games. Not only there's more and more people playing Web games every year, but standards and browsers improve every year in functionality and in performance too.
```

Comment:
```suggestion
But don't underestimate the potential of the Web platform. As I explained in broad terms at the talk I gave at GodotCon Boston 2025, the Web has caught up a lot since the days of Flash games. Not only are there more people playing Web games every year, but standards and browsers improve every year in functionality and in performance.
```